src/pynorare/commands/check.py

```diff
@@ -7,16 +7,14 @@ def run(args):
     for i, ds in enumerate(args.api.datasets.values()):
         visited = set()
         args.log.info('checking {0}'.format(ds.id))
-        for colid, column in ds.columns.items():
-            if colid in args.api.annotations[ds.id]:
-                uniq = '-'.join([
-                    column.language,
-                    column.norare,
-                    column.type,
-                    column.structure,
-                    column.other,
-                ])
-                if uniq in visited:  # pragma: no cover
-                    args.log.warn('non-unique value {0} in {1} / {2}'.format(uniq, ds.id, colid))
-                else:
-                    visited.add(uniq)
+        for column in ds.variables:
+            uniq = '-'.join([
+                column.language,
+                column.norare,
+                column.type,
+                column.structure,
+                column.other,
+            ])
+            if uniq in visited:  # pragma: no cover
+                args.log.warn('non-unique value {0} in {1} / {2}'.format(uniq, ds.id, column.name))
```

Comment:
@LinguList I'm not exactly sure what this check is supposed to prevent. Currently, it emits a handful of warnings, but I couldn't find any obvious errors.

Reply:
Major purpose was to show us the extent to which we have multiple glosses mapped to the same concept set in particular datasets.

Reply:
It doesn't do that. It only checks a certain combination of variable metadata for uniqeness. The actual mapped data isn't looked at at all.

Reply:
Sorry, yes, you're right: I think we introduced this test when we found that we had duplicated values in the data, so they would not be disambiguated, only by their name, due to errors resulting from, e.g., a false language that was assigned to the variable.

Reply:
Since the check does not yield errors or warnings on the current data, it shows we do well.